.NET/Microsoft.Recognizers.Text.DateTime/Parsers/CJK/BaseCJKDatePeriodParser.cs

```diff
@@ -0,0 +1,1622 @@
+                }
+
+                if (!innerResult.Success)
+                {
+                    innerResult = ParseSeason(er.Text, referenceDate);
+                }
+
+                if (!innerResult.Success)
+                {
+                    innerResult = ParseQuarter(er.Text, referenceDate);
+                }
+
+                if (!innerResult.Success)
+                {
+                    innerResult = ParseDecade(er.Text, referenceDate);
+                }
+
+                if (innerResult.Success)
+                {
+                    if (innerResult.FutureValue != null && innerResult.PastValue != null)
+                    {
+                        innerResult.FutureResolution = new Dictionary<string, string>
+                        {
+                            {
+                                TimeTypeConstants.START_DATE,
+                                DateTimeFormatUtil.FormatDate(((Tuple<DateObject, DateObject>)innerResult.FutureValue).Item1)
+                            },
+                            {
+                                TimeTypeConstants.END_DATE,
+                                DateTimeFormatUtil.FormatDate(((Tuple<DateObject, DateObject>)innerResult.FutureValue).Item2)
+                            },
+                        };
+
+                        innerResult.PastResolution = new Dictionary<string, string>
+                        {
+                            {
+                                TimeTypeConstants.START_DATE,
+                                DateTimeFormatUtil.FormatDate(((Tuple<DateObject, DateObject>)innerResult.PastValue).Item1)
+                            },
+                            {
+                                TimeTypeConstants.END_DATE,
+                                DateTimeFormatUtil.FormatDate(((Tuple<DateObject, DateObject>)innerResult.PastValue).Item2)
+                            },
+                        };
+                    }
+                    else
+                    {
+                        innerResult.PastResolution = innerResult.FutureResolution = new Dictionary<string, string>();
+                    }
+
+                    value = innerResult;
+                }
+            }
+
+            var ret = new DateTimeParseResult
+            {
+                Text = er.Text,
+                Start = er.Start,
+                Length = er.Length,
+                Type = er.Type,
+                Data = er.Data,
+                Value = value,
+                TimexStr = value == null ? string.Empty : ((DateTimeResolutionResult)value).Timex,
+                ResolutionStr = string.Empty,
+            };
+
+            return ret;
+        }
+
+        public List<DateTimeParseResult> FilterResults(string query, List<DateTimeParseResult> candidateResults)
+        {
+            return candidateResults;
+        }
+
+        public int GetYearFromText(Match match)
```

Comment:
Is this implementation exactly the same as in the regular DatePeriodParser?

Reply:
yes, added a @TODO to use the method defined in AbstractYearExtractor.